lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -336,15 +338,30 @@ ParseResult FIRParser::parseToken(FIRToken::Kind expectedToken,
   return emitError(message);
 }
 
-/// Parse a list of elements, terminated with an arbitrary token.
+/// Parse a comma-separated list of elements, terminated with an arbitrary
+/// token.
 ParseResult
 FIRParser::parseListUntil(FIRToken::Kind rightToken,
                           const std::function<ParseResult()> &parseElement) {
+  bool first = false;
 
   while (!consumeIf(rightToken)) {
+    // on the second iteration and thereafter, parse a comma.
+    if (first) {
+      if (parseToken(FIRToken::comma, "expected ','")) {
+        return failure();
+      }
+      if (consumeIf(rightToken)) {
```

Comment:
Can we do this change separately? (trailing comma)

I also am not sold it makes sense everywhere we use this?

For example, a trailing comma in a prim op seems weird.  It's nice in things like bundles/enums/types, but seems a little weird in statements -- intrinsics, variadic things like printf/assert, and such.

At least that's my take right now (WDYT? :grin:) but main point is let's tackle that separately.

And wherever it's added, it should have tests please.

Reply:
Fair point on all accounts.

Reply:
I backed out of this change. Trailing commas are back to disallowed.